File: quiz_app_flutter/lib/data/service/quiz_api.dart
import 'package:dio/dio.dart';
import 'package:hooks_riverpod/hooks_riverpod.dart';

import '../../di/network_module.dart';
import '../model/dashboard/dashboard_dto.dart';
import '../model/quiz/quiz_list_dto.dart';

class QuizApi {
  final Dio _dio;
  QuizApi(this._dio);

  Future<DashboardDto> getDashboard() async {
    final response = await _dio.get("quiz_sets.json");
    return DashboardDto.fromJson(response.data);
  }

  Future<QuizListDto> getQuizSet(String fileName) async {
    final response = await _dio.get(fileName);
    return QuizListDto.fromJson(response.data);
  }
}

final quizApiProvider = Provider<QuizApi>((ref) {
  final dio = ref.watch(dioProvider);
  return QuizApi(dio);
});

File: quiz_app_kmp/shared/src/commonMain/kotlin/com/tnm/quizappkmp/shared/data/remote/QuizApi.kt
package com.tnm.shared.data.remote

import com.tnm.shared.data.dto.DashboardDto
import com.tnm.shared.data.dto.QuizListDto
import io.ktor.client.*
import io.ktor.client.call.*
import io.ktor.client.request.*

class QuizApi(private val client: HttpClient, private val baseUrl: String) {

    suspend fun getDashboard(): DashboardDto =
        client.get("${baseUrl}quiz_sets.json").body()

    suspend fun getQuizSet(fileName: String): QuizListDto =
        client.get("$baseUrl$fileName").body()
}